setup.py

```diff
@@ -43,13 +43,30 @@
 
     BOKEHJS_ACTION="install" pip install .
 '''
-from setuptools import find_packages, setup
+from setuptools import find_namespace_packages, setup
+
+import sys
+from pathlib import Path
+
+ROOT = Path(__file__).resolve().parent
+sys.path.append(str(ROOT))  # avoid depending on `build_meta:__legacy__`
 
 import versioneer
-from _setup_support import INSTALL_REQUIRES, BuildJSCmd, check_python
+from _setup_support import INSTALL_REQUIRES, BuildJSCmd, DevelopWithJs, check_python
+
+try:
+    from setuptools.command.build import build  # future-proof
+except ImportError:
+    from distutils.command.build import build
+
+# run `build_js` first, so files are available as "data files" to `build_py`
+build.sub_commands.insert(0, ('build_js', None))
```

Comment:
This follows the approach described in https://github.com/pypa/setuptools/issues/2591.

Reply:
> In any case, for my FYI, build does not happen during an editable install? Is that why this PR has to overwrite develop?

Yes, this is right, the existing implementation of `develop` (to be changed soon), does not run `build` directly. It just calls `build_ext` directly and assume all other files are already in-place.

Reply:
I am currently working on this PR: https://github.com/pypa/setuptools/pull/3265.
It currently does not include the API for subcommands to specify files for the editable install, but I was planning on tackling this at some point in the weekend.

The process I will follow is:

- Merge the change in a few weeks.
- Make it available for beta-testing/community feedback (I will post on the Python discourse on the packaging thread).
- Release under an "experimental flag" so members of the community that don't follow the discourse can provide feedback.
- Finally remove the "experimental flag".

So this might take some time.